ORSSamples/Program.cs

```diff
@@ -0,0 +1,192 @@
+
+            // Demonstrates ORS features. Archiving using batch currently does not work. 
+            // multipleBlobUpdater(sourceContainerClient, destContainerClient, blobName, blobContent1, numberBlobs);
+            blobUpdater(sourceContainerClient, destContainerClient, blobName, blobContent1);
+            blobUpdater(sourceContainerClient, destContainerClient, blobName, blobContent2);          
+            archiveContainerFiles(destContainerClient, destConnectionString, destContainerName);
+            // archiveContainerFilesUsingBatch(destServiceClient, destContainerClient);
+        }
+
+        /*
+         * Uploads 1000 blobs, then checks status of replication in interval of 1 min. Outputs status as a percentage of blobs completed replication
+         */
+        public static void multipleBlobUpdater(BlobContainerClient sourceContainerClient, BlobContainerClient destContainerClient, String blobName, String blobContent, int numberBlobs)
+        {
+            // Uploading blobs
+            Console.WriteLine("Demonstrating tracking percentage of blobs completed");
+            List<string> blobNames = new List<string>();
+            Console.WriteLine("Uploading " + numberBlobs + " blobs");
+            for (int i = 0; i < numberBlobs; i++)
+            {
+                String newBlobName = i + blobName;
+                BlobClient sourceBlob = sourceContainerClient.GetBlobClient(newBlobName);
+                Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(blobContent));
+                sourceBlob.Upload(stream, true);
+                blobNames.Add(newBlobName);
+            }
+
+            // Check if replication in dest container is finished in interval of 5 min
+            Console.WriteLine("Checking to see if replication finished");
+            while (blobNames.Count > 0)
+            {
+                Thread.Sleep(60000);
+                List<string> repIncomplete = new List<string>();
+                foreach (String currBlobName in blobNames)
+                {
+                    bool status = checkReplicationStatus(sourceContainerClient, currBlobName);
+                    if (!status)
+                    {
+                        repIncomplete.Add(currBlobName);
+                    }
+                }
+                blobNames = repIncomplete;
+                Console.WriteLine("Percentage completed: " + ((numberBlobs - blobNames.Count) / numberBlobs * 100 + "%"));
+                Console.WriteLine((numberBlobs - blobNames.Count) + " out of " + numberBlobs + " blobs have completed");
+            }
+        }
+
+        /*
+         * Checks replication status of given blob in given container
+         */
+        public static bool checkReplicationStatus(BlobContainerClient sourceContainerClient, String blobName)
+        {
+            BlobClient sourceBlob = sourceContainerClient.GetBlobClient(blobName);
+            Response<BlobProperties> source_response = sourceBlob.GetProperties();
```

Comment:
I'm not a huge fan of this because doing a get properties on each blob for however long it takes for the blobs to finish replicating, comes at the cost of the customer because each request to do a get properties adds up over time especially if they have over a 1000 blobs. The only way I can think of improving this is to not to do a get properties call on blobs that already have completed (e.g. taking the blob out of the list of blobs we are checking for status)

Maybe we shouldn't do a progress tracker, it just doesn't seem cost efficient at the end of the day. I'm open to discussion about this.

Reply:
I'm currently doing this by keeping track of incomplete blobs in the list "blobNames" and updating it every time I iterate through the while loop. However, I do agree that it doesn't seem very cost efficient to do the progress tracker.

Reply:
Ah I see now! Yeah I think I read over it too quick my bad